core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.spec.ts

```diff
@@ -581,6 +633,36 @@ describe('OptimizedSsrEngine', () => {
         expect(engineRunner.renders).toEqual(['', 'a-1']);
       }));
     });
+
+    it('should render fallback to CSR when rendering request fails', fakeAsync(() => {
+      const engineRunner = new TestEngineRunner(
+        {
+          renderingStrategyResolver: () => RenderingStrategy.ALWAYS_SSR,
+          timeout: 50,
+        },
+        defaultRenderTime,
+        true
+      );
+      engineRunner.request('a');
+      tick(200);
+      expect(engineRunner.renders).toEqual([new Error('a-0')]);
+    }));
```

Comment:
This spec body doesn't match the description - the spec body doesn't test fallbacking to CSR.
Instead, this spec body checks when ALWAYS_SSR is enabled, whether the regular `timeout` will be ignored, and eventually the errored render will be returned.

Suggestion to change the description:
> _should return error response for the errored render, even when the regular timeout passed_